module/app/ante/ante.go

```diff
@@ -1,66 +1,65 @@
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	sdkante "github.com/cosmos/cosmos-sdk/x/auth/ante"
 	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
 	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
 	feegrantkeeper "github.com/cosmos/cosmos-sdk/x/feegrant/keeper"
 	ibcante "github.com/cosmos/ibc-go/v4/modules/core/ante"
 	ibckeeper "github.com/cosmos/ibc-go/v4/modules/core/keeper"
+
+	ethermintante "github.com/evmos/ethermint/app/ante"
+
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/gravity/keeper"
 )
 
 // NewAnteHandler Constructs a new sdk.AnteHandler for the Gravity app.
 // AnteHandlers are functions which validate Txs before their contained Msgs are executed, AnteHandlers are constructed
 // from a chain of AnteDecorators, the final AnteDecorator in the chain being sdkante.Terminator.
-// This custom AnteHandler constructor loosely chains together the pre-Terminator-ed default sdk AnteHandler
-// with additional AnteDecorators. This complicated process is desirable because:
-// 1. the default sdk AnteHandler can change on any upgrade (so we do not want to have a stale list of AnteDecorators),
-// 2. it is not possible to modify an AnteHandler once it is constructed
+// This custom AnteHandler constructor chains together the base CosmosSDK decorators with some other custom decorators,
+// in order to support EIP-712 signed messages (enabling MetaMask signed transactions in browsers) and maintain existing
+// functionality.
+//
+// For more information see https://github.com/evmos/ethermint/blob/v0.19.3/app/ante/ante.go, which inspired this handler.
 func NewAnteHandler(
 	options sdkante.HandlerOptions,
 	gravityKeeper *keeper.Keeper,
 	accountKeeper *authkeeper.AccountKeeper,
 	bankKeeper *bankkeeper.BaseKeeper,
 	feegrantKeeper *feegrantkeeper.Keeper,
 	ibcKeeper *ibckeeper.Keeper,
-	cdc codec.BinaryCodec,
+	cdc codec.Codec,
+	evmChainID string,
 ) (*sdk.AnteHandler, error) {
-	// Call the default sdk antehandler constructor to avoid auditing our changes in the future
-	baseAnteHandler, err := sdkante.NewAnteHandler(options)
-	if err != nil {
-		return nil, sdkerrors.Wrap(err, "Unable to create baseAnteHanlder")
+	if evmChainID == "" {
+		return nil, fmt.Errorf("evmChainID not specified, EIP-712 signing will fail")
 	}
 
-	// Create additional AnteDecorators to chain together
-	ibcAnteDecorator := ibcante.NewAnteDecorator(ibcKeeper)
-	minCommissionDecorator := NewMinCommissionDecorator(cdc)
+	fullHandler := sdk.ChainAnteDecorators(
+		ethermintante.RejectMessagesDecorator{}, // Do not support EVM transactions, easy mistake to make when using MetaMask
+		sdkante.NewSetUpContextDecorator(),
+		// Allows exactly 1 type of extension options, and only one of that type to be provided
+		NewGravityRejectExtensionsDecorator(cdc),
+		sdkante.NewMempoolFeeDecorator(),
+		sdkante.NewValidateBasicDecorator(),
+		sdkante.NewTxTimeoutHeightDecorator(),
+		sdkante.NewValidateMemoDecorator(accountKeeper),
```

Comment:
I think some good tests we could add would be:
1. Check minimum commission violations are rejected
2. Check regular SDK txs and EIP712 txs are accepted (just added)
3. Reject txs with odd extension options or more than 1 web3 extension option (Just added the more than 1 check)
4. Reject EVM txs (e.g. ERC20 transfer) which should fail since Gravity doesn't run an EVM